app/src/main/java/org/wikipedia/edit/richtext/SyntaxHighlighter.kt

```diff
@@ -0,0 +1,250 @@
+                        syntaxHighlightListener?.syntaxHighlightResults(result)
+
+                        // TODO: probably possible to make this more efficient...
+                        // Right now, on longer articles, this is quite heavy on the UI thread.
+                        // remove any of our custom spans from the previous cycle...
+                        var time = System.currentTimeMillis()
+                        val prevSpans = textBox.text.getSpans(0, textBox.text.length, SpanExtents::class.java)
+                        for (sp in prevSpans) {
+                            textBox.text.removeSpan(sp)
+                        }
+                        val findTextList: MutableList<SpanExtents> = ArrayList()
+
+                        // and add our new spans
+                        for (spanEx in result) {
+                            textBox.text.setSpan(spanEx, spanEx.start, spanEx.end, Spanned.SPAN_INCLUSIVE_INCLUSIVE)
+                            if (spanEx.syntaxRule.spanStyle == SyntaxRuleStyle.SEARCH_MATCHES) {
+                                findTextList.add(spanEx)
+                            }
+                        }
+                        if (searchText != null && searchText!!.isNotEmpty() && syntaxHighlightListener != null) {
+                            syntaxHighlightListener!!.findTextMatches(findTextList)
+                        }
+                        time = System.currentTimeMillis() - time
+                        d("That took " + time + "ms")
+                    }) { L.e(it) })
+        }
+    }
+
+    fun applyFindTextSyntax(searchText: String?, listener: OnSyntaxHighlightListener?) {
+        this.searchText = searchText
+        syntaxHighlightListener = listener
+        setSelectedMatchResultPosition(0)
+        postHighlightCallback()
+    }
+
+    fun setSelectedMatchResultPosition(selectedMatchResultPosition: Int) {
+        this.selectedMatchResultPosition = selectedMatchResultPosition
+        postHighlightCallback()
+    }
+
+    private fun postHighlightCallback() {
+        // queue up syntax highlighting.
+        // if the user adds more text within 1/2 second, the previous request
+        // is cancelled, and a new one is placed.
+        handler.removeCallbacks(syntaxHighlightCallback)
+        handler.postDelayed(syntaxHighlightCallback, if (TextUtils.isEmpty(searchText)) DateUtils.SECOND_IN_MILLIS / 2 else 0)
+    }
+
+    fun cleanup() {
+        handler.removeCallbacks(syntaxHighlightCallback)
+        textBox.text.clearSpans()
+        disposables.clear()
+    }
+
+    private inner class SyntaxHighlightTask constructor(private val text: Editable) : Callable<MutableList<SpanExtents>> {
+        private var cancelled = false
+
+        fun cancel() {
+            cancelled = true
+        }
+
+        override fun call(): MutableList<SpanExtents> {
+            val spanStack = Stack<SpanExtents>()
+            val spansToSet: MutableList<SpanExtents> = ArrayList()
```

Comment:
Would it be possible to change to ` val spansToSet = mutableListOf<SpanExtents>()`?